Merge current project into master

Enteris opened the pull request:
Current project is functionnal.

mathi182 committed:
Added automatisation and meta scripts

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    print("     [-d, --duration <HH:MM:SS>]        Maximum duration of the simulation. Default = 24:00:00. Cannot exceed 48h")
+    print("     [-o, --options <option>]           Options for SCHNAPS. See its doc for more information.")
+    print("     [-h, --help]                       Shows the help page\n")
+    print("For help about the RSA key : http://doc.fedora-fr.org/wiki/SSH_:_Authentification_par_cl%C3%A9")
```
Aide en anglais et lien en francais?

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    print("     -u, --username <name>              Username on Colosse for the ssh connection")
+    print("     -p, --project <name>               Project's name")
+    print("     -i, --id <job's id>                Job's id given by Colosse")
+    print("     [-e, --email <address@ulaval.ca>]  Person to join when the simulation is done")
```
Person to ~join~ contact when the simulation is done

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    print("     -p, --project <name>               Project's name")
+    print("     -i, --id <job's id>                Job's id given by Colosse")
+    print("     [-e, --email <address@ulaval.ca>]  Person to join when the simulation is done")
+    print("     [-c]                               Create a new cron job")
```
La création de la cron job est un comportemnt par defaut. Ce param la devrais l'empecher vu qu'il est toujours utilisé par defaut.

mathi182 replied:
C'est l'inverse. Le comportement par défaut de cette partie de script là est d'exécuter une cron job. Y'a juste quand on lance une simulation qu'on demande à créer une cron job.

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    print("     -t, --task                         Index of the jobs' array. Represents the Xe job.")
+    print("     -r, --rap-id                       Rap id.")
+    print("     -s, --scenario                     List of scenarios splitted by the argument.")
+    print("     [-o, --options <option>]           Options for SCHNAPS. See its doc for more information.")
```
Ajoute le lien vers la doc de SCHNAPS (le wiki du github)

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    print("\n" + launcherScript + " :")
+    showHelpLauncher()
+    print("\n" + fetcherScript + " :")
+    showHelpFetcher()
```
Inverser Launcher et Runner de facon a avoir Launcher dans le bas de la fenetre.

mathi182 committed:
Added path to config file

mathi182 committed:
Configuration files are now moved into an archive inside the project folder

mathi182 committed:
Doesn't override the creation date if the meta file already exists

mathi182 committed:
Scripts for compressing configuration files of projects into an archive

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+First script.
+Used by the user.
+It starts the chain by creating a job on the supercalculator, executing it and
+calling the 3rd script on the recieving server.
```
~recieving~ receiving

mathi182 replied:
J'ai vraiment de la misère avec l'ordre de ces lettres là... Receive, Retrieve...

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+    
+    #Accepted arguments
+    try:
+        options, arguments = getopt.getopt(args, "hlp:u:e:d:o:m:", ["help", "log", "project=", "username=", "email=", "duration=", "options=", "mode="])
```
Il faudrais ajouter l'option pour setter la qté de memoire utilisé par la simulation, afin de pouvoir embarquer sur des nodes plus volumineuses (genre les 48Gb)

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+        # 1 job per simulation (Ex. 5 scenarios with 100 simulations = 500 jobs)
+        nbTasks = len(scenarios) * nbIterations
+    elif mode == 2:
+        # 1 job per iteration
```
Ajouter l'exemple : (Ex. 5 scenario with 100 iterations = 100 jobs)

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+        # 1 job per iteration
+        nbTasks = nbIterations
+    elif mode == 3:
+        # 1 job per scenario
```
Idem. (Ex. 5 scenarios with 100 iterations = 5 jobs)

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+                           "#PBS -o " + standardOutputFolder + "/" + projectName + "_%I.out\n" #Standard output
+                           "#PBS -e " + errorOutputFolder + "/" + projectName + "_%I.err\n" #Error output
+                           "#PBS -t [0-" + str(nbTasks) + "]%100\n" # Array of jobs. Max 50 jobs at the same time. Can be anything else than 50 (don't know the max)
+                           
```
Si on crois dépasser la qté de memoire utilisé par la node (les nodes ont 3Gb par coeur sur la queue normale, donc a 8 coeur si on depasse 24Gb, il faut changer de node). Les informations sont dispo ici
https://wiki.calculquebec.ca/w/Ex%C3%A9cuter_une_t%C3%A2che/en#tab=tab2

En fait je dit ca mais jvais créer une issue.

Enteris replied:
La rule of thumb utilisé par le passé pour estimer la memoire est : 
amount_ram = ceil(population*5.3/1000);

Enteris replied:
Par contre c'est p-e dans le champ, ma simulation a utilisé en moyenne 2781.75 Mb par coeur à 8 coeur, donc 22254 Mb en moyenne, et avec le calcul ca dit 17000~. Donc a prendre avec un gain de sel.

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,627 @@
+
+    print("Generating the submit script on Colosse.")
+    ssh.exec_command("echo '" + submitScriptContent + "' > " + join(homeUserPath, submitScriptName) + "\n")
+    
```
J'imagine le handling d'erreur ici a été ajouté dans une nouvelle version?

mathi182 replied:
Y'a pas d'handling d'erreurs. Si ça plante à cette étape là, l'usager va en recevoir la stack trace.

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    
+    try:
+        #Accepted arguments
+        options, arguments = getopt.getopt(args, "p:m:t:o:r:i:s:", ["project=", "mode=", "task=", "iterations=", "options=", "rap-id=", "scenario="])
```
Ajouter des options pour le nombre de proc et la memoire?

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+            if stderr:
+                print("Scenario " + scenario + " : " + stderr)
+                return
+    else:
```
Jmetterais un elif avec le mode 4 puis un else avec un erreur, pour éviter de causer un bug dans le futur

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+    ssh.connect(hostname="colosse.calculquebec.ca", username=username, pkey=k)
+    
+    #Looking for active jobs
+    stin, stout, sterr = ssh.exec_command("showq -u $USER\n")
```
Peut-etre utiliser la commande "checkjob" a place, ca pourrais etre plus simple (mais c'est pas necessaire) Ca permet aussi de savoir d'autre info comme la qté de memoire moyenne utilisé pour mettre dans le output.

mathi182 replied:
Je vais me mettre une note là-dessus, mais je vais mettre ça en priorité plus faible.

Enteris commented on Automatisation/runsim.py:
```diff
@@ -0,0 +1,625 @@
+                        archive.add(join(projectPath, projectName, file))
+            
+        except:
+            print("An error has occurred while retreiving the results : " + str(sys.exc_info()[0]))
```
Si on a fait une erreur pendant le retreiving, faudrais p-e pas enlever la cron job.

Enteris commented on Meta file/metaFile.py:
```diff
@@ -0,0 +1,58 @@
+    metaFile = ".meta"    
+    
+    try:
+        options, arguments = getopt.getopt(args, "hp:m:", ["help", "path=", "meta="])
```
Ajouter un argument pour forcer a réécrire le meta deja présent.

mathi182 replied:
C'est-à-dire? On réécrit toujours par dessus le meta déjà présent. Y'a la creation date qu'on ne touche pas par contre.

Enteris commented on Meta file/metaFile.py:
```diff
@@ -0,0 +1,58 @@
+            size, folder = line.split("\t")
+            with open(join(folder, metaFile), "w") as file:
+                file.write("size:" + size + "\n")
+                file.write("creation date:" + today + "\n")
```
P-e regarder pour utiliser la commande non-POSIX pour avoir la "vrai" date de création plutot que mettre la date d'aujourd'hui? Meme celle qui regarde le cTime serais mieux, vu qu'on peut vouloir le rouler sur un dossier deja créé.

Enteris reviewed (commented):
J'ai mis plein de comment, jpeux pas approve vu que c'est ma PR mais tu regardera ca et la mergera, jche que la source a changé depuis en plus.

mathi182 committed:
Added getCreationTime function, but not usable now! Must be used with sudo...
Modified the meta file function -> Doesn't erase the creation time if it already exists.
Also, corrected errors in PR (comments and unsupported modes)

mathi182 committed:
Added PR from Jean-Alexandre.
Script accepts an amount of ram in input.

mathi182 committed:
Cronjob will not have email in parameters if not specified.
